### PR TITLE
🌱 Update Cluster API IPAM version from v1alpha1 to v1beta1

### LINKED ISSUE #2750: Update Cluster API IPAM version from v1alpha1 to v1beta1 in CAPV
**What steps did you take and what happened:**
While working with the CAPV, I noticed that the IPAM version used is still v1alpha1. This is despite the fact that the IPAM types have been promoted to v1beta1 in the cluster-api. Additionally, the first release of the Cluster API IPAM provider in-cluster uses v1beta1, leading to an inconsistency in the versions used.

**What did you expect to happen:**
NA

**Anything else you would like to add:**
Thought it would be beneficial to align the versions used in CAPV with the promoted IPAM types in the cluster-api and the CAPI IPAM provider in-cluster.

**Environment:**

- Cluster-api-provider-vsphere version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by PavanNeerudu**
**What this PR does / why we need it**:

While working with the CAPV, I noticed that the IPAM version used is still v1alpha1. This is despite the fact that the IPAM types have been promoted to v1beta1 in the cluster-api. Additionally, the first release of the Cluster API IPAM provider in-cluster uses v1beta1, leading to an inconsistency in the versions used.

**Which issue(s) this PR fixes**:
Fixes #2750

### CODE REVIEW

**Review by chrischdi — COMMENTED** [review]
/ok-to-test

File: hack/ipclaim-template.yaml
Comment (chrischdi) [inline hack/ipclaim-template.yaml:1]: Can you remove this file? this was used for CI but is not used anymore at all :-)

File: pkg/services/govmomi/ipam/status_test.go
Comment (chrischdi) [inline pkg/services/govmomi/ipam/status_test.go:28]: please rename this to `ipamv1`, maybe worth enforcing this via `.golangci.yml` like we already do for other apis?

**Review by chrischdi — COMMENTED** [review]
Thanks for the PR, looks straight forward :-)

Some smaller nits.

File: test/framework/ip/incluster.go
Comment (chrischdi) [inline test/framework/ip/incluster.go:42]: Maybe we need to keep the old api group for now for the test code, because in CI v1beta1 is not yet deployed (it is only used for IPAM and thus may be not up to date.

Comment (PavanNeerudu) [reply test/framework/ip/incluster.go]: Yes, makes sense. I was just looking at why the tests were failing and how the testing framework is constructed. Thanks!

File: hack/ipclaim-template.yaml
Comment (PavanNeerudu) [reply hack/ipclaim-template.yaml:1]: Removed in the latest patch.

File: pkg/services/govmomi/ipam/status_test.go
Comment (PavanNeerudu) [reply pkg/services/govmomi/ipam/status_test.go]: Yeah, sure. Added a rule to .golangci.yml as well.

**Comment by chrischdi** [comment]
/test pull-cluster-api-provider-vsphere-test-main

Unrelated flake, which gets hopefully fixed in: 

- #2752

**Review by zhanggbj — APPROVED** [review]
Thanks for the PR! 
/lgtm
Waiting for @chrischdi to have another review.

**Comment by chrischdi** [comment]
Because we don't have any e2e test covering ipam: have you been able to give the implementation a try to create a cluster using IPAM?

**Comment by chrischdi** [comment]
We now have a proper test (via #2757) :-)

/test pull-cluster-api-provider-vsphere-e2e-main

**Comment by chrischdi** [comment]
Test is green 🎉 

Thanks for contributing this!

/tide merge-method squash

**Comment by chrischdi** [comment]
/retest

flake

**Pull request merged**
